### ISSUE TITLE
REPL help for unicode character typing doesn't work for 🍰

### ISSUE THREAD

**Issue by oxinabox** [JuliaLang/julia#38365]
I can type 🍰 by types in `\:cake:` + tab.
but it doesn't display that in help.

```
help?> 🍰
search:

Couldn't find 🍰 ...
```

Contrast to:
```
help?> δ
"δ" can be typed by \delta<tab>

search: ...
```

**Comment by melonedo** [comment]
A shallow search in REPL.jl shows that help mode does not search for emojis:
https://github.com/JuliaLang/julia/blob/548801516dde0d3dae1107b6659901e9f6030b62/stdlib/REPL/src/docview.jl#L370-L380

In tab completion, both latex and emoji symbols are considered:
https://github.com/JuliaLang/julia/blob/548801516dde0d3dae1107b6659901e9f6030b62/stdlib/REPL/src/REPLCompletions.jl#L540-L544

However I am not sure whether they are the very functions that are resorted to when using REPL.

**Comment by stevengj** [comment]
Fixed by #34034.
